views/js/plugins/controls/navigation/navigationlock.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016-2019 (original work) Open Assessment Technologies SA ;
+ */
+
+define(['taoTests/runner/plugin', 'taoQtiTest/runner/helpers/currentItem'], function (pluginFactory, currentItemHelper) {
+    'use strict';
+
+    /**
+     * Returns the configured plugin
+     */
+    return pluginFactory({
+        name: 'navigationlock',
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init: function init() {
+            const testRunner = this.getTestRunner();
+
+            testRunner.after('renderitem', (loadedItem, item) => {
+                const { pci } = item && item.portableElements;
+
+                if (!Object.keys(pci).length) {
+                    return;
+                }
+
+                const itemRunner = testRunner.itemRunner;
+                const elementsOnPage = itemRunner.getData().body.elements;
+
+                const PCINavigationLockExists = {};
+                Object.values(elementsOnPage).forEach(elem => {
+                    if (elem.qtiClass === 'customInteraction' && elem.properties && elem.properties.navigationLock && elem.attributes.responseIdentifier) {
+                        PCINavigationLockExists[elem.attributes.responseIdentifier] = false;
+                    }
+                });
```

Comment:
**thought:** Ok, this shows how to detect the PCI requiring the navigationLock feature. In the end, this must go in a helper as per [TR-4051](https://oat-sa.atlassian.net/browse/TR-4051).